src/controller/java/AndroidLogDownloadFromNode.cpp

```diff
@@ -217,22 +217,27 @@ void AndroidLogDownloadFromNode::FinishLogDownloadFromNode(CHIP_ERROR err)
         // Java method signature : boolean onSuccess(int fabricIndex, long nodeId)
         jniErr = JniReferences::GetInstance().FindMethod(env, mJavaCallback.ObjectRef(), "onSuccess", "(IJ)V", &onSuccessMethod);
 
-        VerifyOrReturn(jniErr == CHIP_NO_ERROR, ChipLogError(Controller, "Could not find onSuccess method"));
+        VerifyOrExit(jniErr == CHIP_NO_ERROR, ChipLogError(Controller, "Could not find onSuccess method"));
 
         env->CallVoidMethod(mJavaCallback.ObjectRef(), onSuccessMethod, static_cast<jint>(mController->GetFabricIndex()),
                             static_cast<jlong>(mRemoteNodeId));
-        return;
     }
+    else
+    {
+        ChipLogError(Controller, "Log Download Failed : %" CHIP_ERROR_FORMAT, err.Format());
 
-    ChipLogError(Controller, "Log Download Failed : %" CHIP_ERROR_FORMAT, err.Format());
+        jmethodID onErrorMethod;
+        // Java method signature : void onError(int fabricIndex, long nodeId, long errorCode)
+        jniErr = JniReferences::GetInstance().FindMethod(env, mJavaCallback.ObjectRef(), "onError", "(IJJ)V", &onErrorMethod);
+        VerifyOrExit(jniErr == CHIP_NO_ERROR, ChipLogError(Controller, "Could not find onError method"));
 
-    jmethodID onErrorMethod;
-    // Java method signature : void onError(int fabricIndex, long nodeId, long errorCode)
-    jniErr = JniReferences::GetInstance().FindMethod(env, mJavaCallback.ObjectRef(), "onError", "(IJJ)V", &onErrorMethod);
-    VerifyOrReturn(jniErr == CHIP_NO_ERROR, ChipLogError(Controller, "Could not find onError method"));
+        env->CallVoidMethod(mJavaCallback.ObjectRef(), onErrorMethod, static_cast<jint>(mController->GetFabricIndex()),
+                            static_cast<jlong>(mRemoteNodeId), static_cast<jlong>(err.AsInteger()));
+    }
 
-    env->CallVoidMethod(mJavaCallback.ObjectRef(), onErrorMethod, static_cast<jint>(mController->GetFabricIndex()),
-                        static_cast<jlong>(mRemoteNodeId), static_cast<jlong>(err.AsInteger()));
+exit:
+    // Finish this function, this object will be deleted.
+    delete this;
```

Comment:
this is not correct, this function is inside callback, you cannot delete this, you should pass context object(AndroidLogDownloadFromNode) from the  callback inside FinishLogDownloadFromNode, then delete in the end.

Reply:
Modified. 
And. The code where Memory Leakage occurred was confirmed to be not Java, but all-cluster-app. However, I checked that there was a part where all-cluster-app was terminated quickly while the script was running, and memory was not freed. To prevent this, I added sleep in the java controller as follows.
https://github.com/project-chip/connectedhomeip/pull/36742/files#diff-bd1929217ea9054b007b99dc0e924dfec04dfbb662e4d00e8c98c9b03fea8901R89